src/components/App.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from "react";
+import Search from "./Search";
+import SearchResults from "./SearchResults";
+import Popup from "./Popup";
+import Slideshow from "./Slideshow";
+import "../styles/app.css";
+
+const App = () => {
+  const [fetchData, setFetchData] = useState();
+  const [errorMessage, setErrorMessage] = useState();
+  const [buttonPopup, setButtonPopup] = useState(false);
+  const [src, setSrc] = useState({ href: "", title: "" });
+  const [slideOn, setSlideOn] = useState(false);
+
+  const handleClick = (event) => {
+    setSrc({ href: event.target.src, title: event.target.title });
+    setButtonPopup(true);
+  };
+
+  useEffect(() => {
+    const images = Array.from(document.getElementsByClassName("image"));
+    images.forEach((image) => {
+      image.addEventListener("click", handleClick);
+    });
+
+    return () => {
+      images.forEach((image) => {
+        image.removeEventListener("click", handleClick);
+      });
+    };
+  }, [fetchData, slideOn]);
+
+  return (
+    <div className="app">
+      {fetchData && fetchData.length > 0 && (
+        <button
+          className="clear-mode"
+          type="button"
+          onClick={() => {
+            setFetchData([]);
+            setButtonPopup(false);
+            setSrc({ href: "", title: "" });
+            setSlideOn(false);
+          }}
```

Comment:
Id move this into a separate function above the return statement so you can keep the JSX a bit cleaner